### PR TITLE
HSDir Store Events (Feature Complete)

### PR DESCRIPTION
**Pull request by teor2345**
I have working event code, that contains all the fields @robgjansen and @ajohnson-nrl  have asked for.

From the release notes privcount/privcount@2db96dc:
* Add the PRIVCOUNT_HSDIR_CACHE_STORED event to the PrivCount Tor Patch #336
* Use tagged fields for new events #256
* Increase the maximum tor event length to 2kB #336

For the fields in the event, see privcount/privcount@be59e55.
I still have to do testing locally and on a relay.

### CODE REVIEW

File: src/or/rendcache.c [hunk 1]
```diff
@@ -696,9 +720,59 @@ rend_cache_store_v2_desc_as_dir(const char *desc)
...
+      /* Successful stores */
+      tor_assert(!failure_reason);
+      control_event_privcount_hsdir_cache_stored(HS_VERSION_TWO,
+                                               /* cache info */
+                                               has_existing_cache_entry,
+                                               1, /* added to cache */
+                                               (has_existing_cache_entry
+                                                ? "updated"
+                                                : "new"),
+                                               /* descriptor info */
```
Comment (robgjansen) [inline src/or/rendcache.c:744]: I don't understand this coding style of having so much space to the left of separate parameters in a function block that each parameter has to be split across multiple lines. Why not just decrease the indent, so at least the result would be one parameter+comment per line? I realize that the parameters are lined up with the end of the function name and that this stile may be used elsewhere in Tor as well; but the style is very visually unappealing to me. (This is not a big deal, just my opinion.)

**Review by robgjansen — COMMENTED** [review]
Looks fine to me!

File: src/or/rendcache.c (hunk 1 above)
Comment (teor2345) [reply src/or/rendcache.c:744]: I agree it's ugly, it's a compromise between make check-spaces and my editor's automatic indenting. I'll try to do better next time.

**Comment by teor2345** [comment]
This was squashed and merged to privcount-master in 19c6ea9cf.

**Pull request closed without merging**